addons/controllers/webhooks_test.go

```diff
@@ -0,0 +1,155 @@
+		It("should install new certificates", func() {
+			testTLSSecret := webhooks.WebhookTLS{
+				Ctx:           ctx,
+				K8sConfig:     k8sConfig,
+				CertPath:      certPath,
+				KeyPath:       keyPath,
+				Name:          webhookScrtName,
+				ServiceName:   webhookServiceName,
+				LabelSelector: "webhook-cert=self-managed",
+				Logger:        setupLog,
+				TLSSecret:     nil,
+				Namespace:     addonNamespace,
+				InvalidWithIn: time.Hour * 24,
+			}
+			err := testTLSSecret.Manage()
+			Expect(err).ToNot(HaveOccurred())
+			Expect(err).ToNot(HaveOccurred())
+			Expect(testTLSSecret.TLSSecret).NotTo(BeNil())
+			cert, err := cert2.CertsFromFile(certPath)
+			Expect(err).ToNot(HaveOccurred())
+			certPEM1, err := cert2.EncodeCertificates(cert[0])
+			Expect(err).ToNot(HaveOccurred())
+			Expect(certPEM1).To(Equal(testTLSSecret.TLSSecret.Data[resources.ServerCert]))
+			key, err := keyutil.PrivateKeyFromFile(keyPath)
+			Expect(err).ToNot(HaveOccurred())
+			orgKey, err := keyutil.ParsePrivateKeyPEM(testTLSSecret.TLSSecret.Data[resources.ServerKey])
+			Expect(err).ToNot(HaveOccurred())
+			Expect(key).To(Equal(orgKey))
+
+			err = testTLSSecret.Manage()
+			Expect(err).ToNot(HaveOccurred())
+			Expect(err).ToNot(HaveOccurred())
+			Expect(testTLSSecret.TLSSecret).NotTo(BeNil())
+			cert, err = cert2.CertsFromFile(certPath)
+			Expect(err).ToNot(HaveOccurred())
+			certPEM2, err := cert2.EncodeCertificates(cert[0])
+			Expect(err).ToNot(HaveOccurred())
+			Expect(certPEM2).To(Equal(testTLSSecret.TLSSecret.Data[resources.ServerCert]))
+			key, err = keyutil.PrivateKeyFromFile(keyPath)
+			Expect(err).ToNot(HaveOccurred())
+			orgKey, err = keyutil.ParsePrivateKeyPEM(testTLSSecret.TLSSecret.Data[resources.ServerKey])
+			Expect(err).ToNot(HaveOccurred())
+			Expect(key).To(Equal(orgKey))
+			Expect(certPEM1).To(Equal(certPEM2))
+		})
+	})
+})
+
+var _ = Describe("when TLS certificates are being managed", func() {
+	Context("TLS certificate Management", func() {
+		It("should continuously update and validate certificates ", func() {
+			privCtx := context.Background()
+			cacelCtx, cancelFn := context.WithCancel(privCtx)
+			invalidTime := (time.Hour * 24 * 7) - (time.Second * 10)
+			webhookTLS := webhooks.WebhookTLS{
+				Ctx:           cacelCtx,
+				K8sConfig:     k8sConfig,
+				CertPath:      certPath,
+				KeyPath:       keyPath,
+				Name:          webhookScrtName,
+				ServiceName:   webhookServiceName,
+				LabelSelector: "webhook-cert=self-managed",
+				Logger:        setupLog,
+				TLSSecret:     nil,
+				Namespace:     addonNamespace,
+				InvalidWithIn: invalidTime,
+			}
+
+			go webhooks.ManageTLSCertificates(&webhookTLS, time.Second*10)
+			time.Sleep(time.Second * 10)
+			firstCert, err := cert2.CertsFromFile(webhookTLS.CertPath)
+			Expect(err).ToNot(HaveOccurred())
+			firstCertPEM, err := cert2.EncodeCertificates(firstCert[0])
+			Expect(err).ToNot(HaveOccurred())
+
+			time.Sleep(time.Second * 20)
```

Comment:
question: trying to understand why 20sec here, would it be a good idea to have some comments ?

Reply:
Here I am jus trying to make sure we don't pickup the same certificate. 
The MageneTLSCertificate is configured to rotate the certificate every 10 seconds. 
I guess we could make this 15 seconds, in theory any number larger than 10 would work, but 20 is a good buffer I think

Reply:
Made tests derive their "wait" times from the rotation time provided for certificates.